Rest/Controller/UserController.php

```
@@ -200,22 +236,52 @@ public function postAction(Request $request)
         }
 
         $user = $this->deserializeEntity($request->request->all(), $user);
-
         // handle the password
         if ($request->request->has('password')) {
-            $encoderFactory = $this->getContainer()->get('security.context')->getEncoderFactory();
-            $password = $request->request->get('password');
+           $password = $request->request->get('password');
+        } else {
+            $password = substr(hash('sha512', rand()), 0, 6);
```

o_O if user don't return password, you automaticaly set a generated ? imo you should return an exception here.